Generate file hashes from Etag information

IchHabRecht opened the pull request:
Instead of downloading a file to be able to generate its content hash,
the Etag information from S3 API is used to generate the according hash.
This improves performance a lot as only a head request to the S3 API is
needed. No file has to be downloaded to the local storage anymore.
According to the S3 API, the Etag information changes when the content
of a file changes. Thus it is save to rely on this information. As two
hash methods (sha1 and md5) need to be supported, the Etag information
is used to be hashed again, using one of the proper hash functions. This
ensures that the generated hash is a valid sha1 and/or md5 hash, that
can be validated furthermore.

Changing the hash method, the available hashes for existing files (and
their processed files) need to be regenerated. Therefore an update
wizard is introduced that can regenerate bulk information and update
hashes in the database directly.

IchHabRecht committed:
[FEATURE] Generate file hashes from Etag information

Instead of downloading a file to be able to generate its content hash,
the Etag information from S3 API is used to generate the according hash.
This improves performance a lot as only a head request to the S3 API is
needed. No file has to be downloaded to the local storage anymore.
According to the S3 API, the Etag information changes when the content
of a file changes. Thus it is save to rely on this information. As two
hash methods (sha1 and md5) need to be supported, the Etag information
is used to be hashed again, using one of the proper hash functions. This
ensures that the generated hash is a valid sha1 and/or md5 hash, that
can be validated furthermore.

Changing the hash method, the available hashes for existing files (and
their processed files) need to be regenerated. Therefore an update
wizard is introduced that can regenerate bulk information and update
hashes in the database directly.

manuelselbach commented on Classes/Driver/AmazonS3Driver.php:
```diff
@@ -617,26 +617,24 @@ public function deleteFile($fileIdentifier)
+
+        $basePath = '';
+        if (!empty($this->configuration['basePath'])) {
+            $basePath = '/' . trim($this->configuration['basePath'], '/');
```
Do we need to add the slash here? AFAICS, the leading slash will be removed anyway and `$basePath` is not used anywhere else.

manuelselbach commented on Classes/Updates/Sha1Update.php:
```diff
@@ -0,0 +1,185 @@
+     */
+    public function checkForUpdate(&$description)
+    {
+        $description = 'The hash calculation for files (content) has been changed with.'
```
I guess the `with` at the end was added by accident here, right? 😁

IchHabRecht committed:
[BUGFIX] Remove superfluous slashes on basePath

In both cases the basePath is used as key for an api call and trims slashes anyway.

IchHabRecht committed:
[BUGFIX] Correct UpdateWizard description
